scripts/migration/migrate-addon.sh

```diff
@@ -0,0 +1,653 @@
+3521f045eda95b55b2bab7922bf7eeeec08a00d9662eb08863beadd82b246289
+46319a44f52ebf396c6fa86548aed55fa691c11daf1a0242380d3f119254fd81
+48011addf964102571fccb1a56010a10c8eadc5fe96c47e7cdf207ebdac61ad9
+55a0a1f2325399064d70c332af5848e672c3beb582ac48546c65231b7a822555
+633ae8afd07bada4fdc8f140cc4bc59d884f2204d47f813086c68c441d87dec8
+660943ace01de1e530322c8091fea858d4459ee2992eba4e1368a686e8f9ad62
+66cd7d3c22417ec760ba69dfdf251a98ab09498d8eed4521081487ac99a9f114
+70764778c47cfda7a55ba4b8d83c696b3ea1c97483564008a15930407e36ecee
+7acac4740c8ec591d110700eb29c6eb54248f2f89ee5db184257564d3438eb8c
+7d7e83fc25d4d8057c2eaaed4553f8692b217e24cecd111382456004913df937
+8034f38e32074df7236d37c37e34e233553d9ab3ddff7f1fd13f23cf14a57689
+8c27b82610b26de3630d025a6028d9dafb242452979588be336da9d40c68b9ac
+a01ef82a0d94488bfa2863bd43a56f140d76b22b3d510190b4afb7fdab377365
+bdf825d61fc924840283c52249dcbefa5708ee97a6d0878cec1126528af6f4d9
+c83d97ab08807f8c8feb706ad0349eca5d510b109ef1824eb1a9799a2f8bae74
+cbdfee7d86bc88f9b20cf9d6525cf2c817fb9e31768fd7484bfe7ee0e8416b13
+cc3e9d8588165999a6bb3ecbe3433b91c77c7a31532f8e5a12871801b3d9545b
+d471a464930c5c575f2371775311547216ea1c735059c1ed14d81fa00262d572
+e24cef4a64476c183257dbbb96951676368487c9230fd6613a98f18c6299f255
+e4789137cf4b070c4ba0b6b71cfba410dd863dbdf06dd4095bc8e1ca2e1ad1b0
+e5e6971d37fa0e98e5e95ee3c9c0f53d3396105022f56e2ba22cf2bf4805be8a
+e6de2153cd51ca711a8fc56b515fd51446188817bae76e67581c8bc8d388699f
+ef8588aff68f7a6c2ea3bad7283e70da3dfaf04b33f169176bbfa8023a02a998
+efb2f225194335ceffb158f73bd14ac85c7bb473797d75914179660c9c2db967
+facc7fd2599994a7c92a4cc66bac9061b64d71be106e2bb639066969273d4fb8"
+  )
+  CUSTOM_MIXER="false"
+  for type in handler rule handler; do
+    names=$(kubectl get "${type}" -oname -n istio-system)
+    for rn in ${names}; do
+      hd=$(kubectl get "${rn}" -n istio-system -o jsonpath='{.spec}')
+      actualSHA=$(echo "${hd}" | tr -d '\n' | sha256sum | awk '{print $1}')
+      builtinSHA=${rsHashMap[${type}]}
+      if [[ $builtinSHA != *"${actualSHA}"* ]];then
+        echo "custom mixer resource found, resource type: ${type}, name: ${rn}"
+        CUSTOM_MIXER="true"
+      fi
+    done
+  done
+  if [[ "${CUSTOM_MIXER}" == "true" ]];then
+    return 1;
+  fi
+}
+
+config_check() {
+
+  download_and_install_migration_tool
+
+  local LAST_ERR;
+  heading "Checking for configurations that will need to be explicitly migrated..."
+
+  # EnvoyFilter
+  local FILTERS
+  FILTERS=$(kubectl get envoyfilter -A -l 'operator.istio.io/component!=Pilot')
+  if [[ "${FILTERS}" != "" ]]; then
+    prompt_unsupported "Custom Envoy Filters are not supported by ASM. Please remove these if possible" "${FILTERS}"
+  fi
+
+  # Plugin Cert
+  local PLUGINCERT
+  PLUGINCERT=$(kubectl get secret cacerts -n istio-system --ignore-not-found)
+  if [[ "${PLUGINCERT}" != "" ]]; then
+    prompt_unsupported "Detected custom plugin cert" "The plugin certs will not be migrated to ASM.."
```

Comment:
Get rid of extra period at the end. Also, the error messages aren't consistent with punctuation or capitalization.